pkg/applicationserver/io/web/webhooks.go

```diff
@@ -394,21 +396,25 @@ func (w *webhooks) newRequest(ctx context.Context, msg *ttnpb.ApplicationUp, hoo
 	if err != nil {
 		return nil, err
 	}
-	req, err := http.NewRequest(http.MethodPost, url.String(), bytes.NewReader(buf))
-	if err != nil {
-		return nil, err
-	}
-	for key, value := range hook.Headers {
-		req.Header.Set(key, value)
-	}
-	if hook.DownlinkAPIKey != "" {
-		req.Header.Set(downlinkKeyHeader, hook.DownlinkAPIKey)
-		req.Header.Set(downlinkPushHeader, w.createDownlinkURL(ctx, hook.ApplicationWebhookIdentifiers, msg.EndDeviceIdentifiers, "push"))
-		req.Header.Set(downlinkReplaceHeader, w.createDownlinkURL(ctx, hook.ApplicationWebhookIdentifiers, msg.EndDeviceIdentifiers, "replace"))
+	var requests []*http.Request
+	for i, v := range buf {
+		req, err := http.NewRequest(http.MethodPost, url.String(), bytes.NewReader(v))
+		requests[i] = req
```

Comment:
This will panic since the slice is empty. Use `requests = append(requests, req)` instead.
Also do the appending at the end of the loop (otherwise you will append values for which errors might exist).